src/main/scala/firrtl/Emitter.scala

```diff
@@ -31,17 +34,68 @@ sealed trait EmitAnnotation extends NoTargetAnnotation {
 case class EmitCircuitAnnotation(emitter: Class[_ <: Emitter]) extends EmitAnnotation
 case class EmitAllModulesAnnotation(emitter: Class[_ <: Emitter]) extends EmitAnnotation
 
+object EmitCircuitAnnotation extends HasShellOptions {
+
+  val options = Seq(
+    new ShellOption[String](
+      longOption = "emit-circuit",
+      toAnnotationSeq = (a: String) => a match {
+        case "chirrtl"              => Seq(RunFirrtlTransformAnnotation(new ChirrtlEmitter),
+                                           EmitCircuitAnnotation(classOf[ChirrtlEmitter]))
+        case "high"                 => Seq(RunFirrtlTransformAnnotation(new HighFirrtlEmitter),
+                                           EmitCircuitAnnotation(classOf[HighFirrtlEmitter]))
+        case "middle"               => Seq(RunFirrtlTransformAnnotation(new MiddleFirrtlEmitter),
+                                           EmitCircuitAnnotation(classOf[MiddleFirrtlEmitter]))
+        case "low"                  => Seq(RunFirrtlTransformAnnotation(new LowFirrtlEmitter),
+                                           EmitCircuitAnnotation(classOf[LowFirrtlEmitter]))
+        case "verilog" | "mverilog" => Seq(RunFirrtlTransformAnnotation(new VerilogEmitter),
+                                           EmitCircuitAnnotation(classOf[VerilogEmitter]))
+        case "sverilog"             => Seq(RunFirrtlTransformAnnotation(new SystemVerilogEmitter),
+                                           EmitCircuitAnnotation(classOf[SystemVerilogEmitter]))
+        case _                      => throw new PhaseException(s"Unknown emitter '$a'! (Did you misspell it?)") },
+      helpText = "Run the specified circuit emitter (all modules in one file)",
+      shortOption = Some("E"),
+      helpValueName = Some("<chirrtl|high|middle|low|verilog|mverilog|sverilog>") ) )
+
+}
+
+object EmitAllModulesAnnotation extends HasShellOptions {
+
+  val options = Seq(
+    new ShellOption[String](
+      longOption = "emit-modules",
```

Comment:
I can't seem to find where `--emit-modules ` was deleted but can you add a better message for when people try to use it, similarly to what you did with `--top-name`?

Reply:
Just to clarify, you're talking about `--split-modules` yes?

Reply:
Yeah sorry, updated

Reply:
Fixed in rebase. This includes a test, like with `--top-name/-tn`.